Bootstrap Errors from Rust Clippy

## Describe the bug
Running bootstrap.ps1 results in errors that prevent the simulators from being built properly. This appears to be related to a new release of rust clippy adding new warnings.

## To Reproduce
Clone qsharp-runtime repo
Install listed pre-requisites and Powershell 7
Run boostrap.ps1

## Expected behavior
boostrap runs without errors

## Actual behavior
Errors from rust clippy (pasted below)

## System information
OS: Windows
rustc/rust-clippy version: 1.55.0

## Error messages

```
Compiling qdk_sim_experimental v0.0.1-alpha
Checking num_enum v0.5.4
error[E0658]: custom inner attributes are unstable
 --> src/lib.rs:4:4
  |
4 | #![clippy::allow(needless_borrow)] // because of strange rust clippy error
  |    ^^^^^^^^^^^^^
  |
  = note: see issue #54726 <https://github.com/rust-lang/rust/issues/54726> for more information

error: usage of unknown attribute
 --> src/lib.rs:4:12
  |
4 | #![clippy::allow(needless_borrow)] // because of strange rust clippy error
  |            ^^^^^

For more information about this error, try `rustc --explain E0658`.
error: could not compile `qdk_sim_experimental` due to 2 previous errors

...
Compiling qdk_sim_experimental v0.0.1-alpha 
...

error[E0658]: custom inner attributes are unstable
 --> src/lib.rs:4:4
  |
4 | #![clippy::allow(needless_borrow)] // because of strange rust clippy error
  |    ^^^^^^^^^^^^^
  |
  = note: see issue #54726 <https://github.com/rust-lang/rust/issues/54726> for more information
  = help: add `#![feature(custom_inner_attributes)]` to the crate attributes 
to enable

For more information about this error, try `rustc --explain E0658`.
error: could not compile `qdk_sim_experimental` due to previous error

...
Compiling qdk_sim_experimental v0.0.1-alpha
Documenting qdk_sim_experimental v0.0.1-alpha 
...

warning: lint `missing_doc_code_examples` has been renamed to `rustdoc::missing_doc_code_examples`
  --> src/lib.rs:22:23
   |
22 | #![cfg_attr(doc, warn(missing_doc_code_examples))]
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^ help: use the new name: 
`rustdoc::missing_doc_code_examples`
   |
   = note: `#[warn(renamed_and_removed_lints)]` on by default

error[E0658]: custom inner attributes are unstable
 --> src/lib.rs:4:4
  |
4 | #![clippy::allow(needless_borrow)] // because of strange rust clippy error
  |    ^^^^^^^^^^^^^
  |
  = note: see issue #54726 <https://github.com/rust-lang/rust/issues/54726> for more information
  = help: add `#![feature(custom_inner_attributes)]` to the crate attributes 
to enable

error: Compilation failed, aborting rustdoc

For more information about this error, try `rustc --explain E0658`.
warning: `qdk_sim_experimental` (lib doc) generated 1 warning
error: could not document `qdk_sim_experimental`
```

--- a/src/Simulation/qdk_sim_rs/src/c_api.rs
+++ b/src/Simulation/qdk_sim_rs/src/c_api.rs
@@ -101,7 +101,7 @@ pub extern "C" fn lasterr() -> *const c_char {
 /// - **`pure`**: Creates the simulator with an initial state represented by
 ///   a state vector.
 /// - **`mixed`**: Creates the simulator with an initial state represented by
-///   a density operat.
+///   a density operator.
 /// - **`stabilizer`**: Creates the simulator with an initial state represented by
 ///   a stabilizer tableau.
 ///

--- a/src/Simulation/qdk_sim_rs/src/processes/apply.rs
+++ b/src/Simulation/qdk_sim_rs/src/processes/apply.rs
@@ -27,9 +27,9 @@ impl Process {
         }
 
         match &self.data {
-            Unitary(u) => apply_unitary(&u, state),
-            KrausDecomposition(ks) => apply_kraus_decomposition(&ks, state),
-            MixedPauli(paulis) => apply_pauli_channel(&paulis, state),
+            Unitary(u) => apply_unitary(u, state),
+            KrausDecomposition(ks) => apply_kraus_decomposition(ks, state),
+            MixedPauli(paulis) => apply_pauli_channel(paulis, state),
             Sequence(processes) => {
                 // TODO[perf]: eliminate the extraneous clone here.
                 let mut acc_state = state.clone();

--- a/src/Simulation/qdk_sim_rs/src/states.rs
+++ b/src/Simulation/qdk_sim_rs/src/states.rs
@@ -162,7 +162,7 @@ impl Trace for &State {
     fn trace(self) -> Self::Output {
         match &self.data {
             Pure(_) | StateData::Stabilizer(_) => C64::one(),
-            Mixed(ref rho) => (&rho).trace(),
+            Mixed(ref rho) => rho.trace(),
         }
     }
 }

